ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/OnMapAndViewReadyListener.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.kotlindemos
+
+import android.annotation.SuppressLint
+import android.os.Build
+import android.view.View
+import android.view.ViewTreeObserver.OnGlobalLayoutListener
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+
+/**
+ * Helper class that will delay triggering the OnMapReady callback until both the GoogleMap and the
+ * View having completed initialization. This is only necessary if a developer wishes to immediately
+ * invoke any method on the GoogleMap that also requires the View to have finished layout
+ * (ie. anything that needs to know the View's true size like snapshotting).
+ */
+class OnMapAndViewReadyListener(
+        private val mapFragment: SupportMapFragment,
+        private val toBeNotified: OnGlobalLayoutAndMapReadyListener
+    ) : OnGlobalLayoutListener,
+        OnMapReadyCallback {
+
+    private val mapView: View? = mapFragment.view
+
+    private var isViewReady = false
+    private var isMapReady = false
+    private var googleMap: GoogleMap? = null
+
+    /** A listener that needs to wait for both the GoogleMap and the View to be initialized.  */
+    interface OnGlobalLayoutAndMapReadyListener {
+        fun onMapReady(googleMap: GoogleMap?)
+    }
+
+    init {
+        registerListeners()
+    }
+
+    private fun registerListeners() {
+        // View layout.
+        if (mapView?.width != 0 && mapView?.height != 0) {
+            // View has already completed layout.
+            isViewReady = true
+        } else {
+            // Map has not undergone layout, register a View observer.
+            mapView.viewTreeObserver.addOnGlobalLayoutListener(this)
+        }
+
+        // GoogleMap. Note if the GoogleMap is already ready it will still fire the callback later.
+        mapFragment.getMapAsync(this)
+    }
+
+    override fun onMapReady(googleMap: GoogleMap) {
```

Comment:
Should googleMap be nullable here?

Reply:
Yes, it should. Thanks for catching that! I've fixed that, added an early return if it is null, and renamed the GoogleMap property in the class to map to avoid the this.googleMap in the next line.